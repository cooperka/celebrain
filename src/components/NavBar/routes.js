import Home from '../Home/component';
import Game from '../Game/component';
import About from '../About/component';

// Note: Make sure to also update static.config routes.
const routes = [
  {
    path: '/',
    component: Home,
    displayName: 'Home',
  },
  {
    path: '/game',
    component: Game,
    displayName: 'Play now',
  },
  {
    path: '/about',
    component: About,
    displayName: 'About',
  },
];

export default routes;
