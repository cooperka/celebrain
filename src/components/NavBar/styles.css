.NavBar {
  /* Default is 64. Specified here to be explicit. */
  height: 64px;
}

.NavBar .buttons {
  flex: 1;
  display: flex;
  justify-content: flex-end;
}

.NavBar .logo {
  height: 44px;
}
