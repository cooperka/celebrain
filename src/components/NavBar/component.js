// @flow

import React from 'react';
import { AppBar, Toolbar } from 'material-ui';

import logo from '../../images/logo.svg';

import NavButton from '../NavButton';

import './styles.css';

function NavBar() {
  return (
    <AppBar color="default">
      <Toolbar className="NavBar">
        <a href="/">
          <img className="logo" src={logo} alt="Home" />
        </a>
        <div className="buttons">
          <NavButton linkTo="/">Home</NavButton>
          <NavButton linkTo="/services">Services</NavButton>
          <NavButton externalLinkTo="mailto:kevin@lsrfocus.com">Contact</NavButton>
        </div>
      </Toolbar>
    </AppBar>
  );
}

export default NavBar;
