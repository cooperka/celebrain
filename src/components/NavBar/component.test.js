// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import NavBar from './component';

it('renders', () => {
  const tree = renderer.create(<NavBar />);
  expect(tree.toJSON()).toMatchSnapshot();
});
