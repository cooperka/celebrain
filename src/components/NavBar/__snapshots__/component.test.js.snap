// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<header
  className="MuiPaper-root-9 MuiPaper-shadow4-15 MuiAppBar-root-1 MuiAppBar-positionFixed-2 MuiAppBar-colorDefault-6 mui-fixed"
>
  <div
    className="MuiToolbar-root-36 MuiToolbar-gutters-37 NavBar"
  >
    <a
      href="/"
    >
      <img
        alt="Home"
        className="logo"
        src="logo.svg"
      />
    </a>
    <div
      className="buttons"
    >
      <NavLink
        className="MuiButtonBase-root-53 MuiButton-root-38 NavButton "
        exact={true}
        href={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        to="/"
      >
        <span
          className="MuiButton-label-39"
        >
          <div
            className="link-text"
          >
            Home
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-55"
        />
      </NavLink>
      <NavLink
        className="MuiButtonBase-root-53 MuiButton-root-38 NavButton "
        exact={true}
        href={undefined}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        to="/about"
      >
        <span
          className="MuiButton-label-39"
        >
          <div
            className="link-text"
          >
            About
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-55"
        />
      </NavLink>
      <a
        className="MuiButtonBase-root-53 MuiButton-root-38 NavButton "
        exact={undefined}
        href="https://github.com/cooperka/celebrain"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        to={undefined}
      >
        <span
          className="MuiButton-label-39"
        >
          <div
            className="link-text"
          >
            GitHub
          </div>
        </span>
        <span
          className="MuiTouchRipple-root-55"
        />
      </a>
    </div>
  </div>
</header>
`;
